common/client/models.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+type ClientErrorType int
+
+// Generalized client error types that dictates what should be the next action, depending on the RPC error response.
+const (
+	Successful              ClientErrorType = iota
+	Fatal                                   // Unrecoverable error. Most likely the attempt should be thrown away.
+	Retryable                               // The error returned by the RPC indicates that if we retry with the same attempt, the tx will eventually go through.
+	Underpriced                             // Attempt was underpriced. New estimation is needed with bumped gas price.
+	Unknown                                 // Tx failed with an error response that is not recognized by the txm.
```

Comment:
nit:  >>"that is not recognized by the txm."

This comment is for a code returned by Client, to the Txmgr. So comment should say something specific to client:

// Tx failed with an error response from RPC, that is not recognized by the client.

Reply:
Same for the next line comment